source/UF/ConnectedTypes.lagda

```diff
@@ -0,0 +1,223 @@
+Ian Ray, 23rd July 2024
+
+We will define connected types and maps (recall our convetion for H-levels starts
+at 0). We then explore relationships, closure properties and characterizations
+of interest pertaining to the concept of connectedness.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+open import UF.PropTrunc 
+
+module UF.ConnectedTypes (fe : Fun-Ext)
+                         (pt : propositional-truncations-exist)
+                          where
+                          
+open import MLTT.Spartan
+open import UF.Base
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.Equiv-FunExt
+open import UF.IdentitySystems
+open import UF.Retracts
+open import UF.Sets
+open import UF.Singleton-Properties
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Subsingletons-Properties
+open import UF.Univalence
+open import UF.UA-FunExt
+open import Naturals.Addition renaming (_+_ to _+'_)
+open import Naturals.Order
+open import UF.ImageAndSurjection pt
+open import UF.H-Levels fe  
+open import UF.Truncations fe pt
+
+\end{code}
+
+We now define the notion of k-connectedness for types and functions with respect
+to H-levels. TODO: show that connectedness as defined elsewhere is a
+special case of k-connectedness. Connectedness typically means set connectedness.
+In terms of our Hlevel convetion it will mean 2-connectedness.
+
+\begin{code}
+
+module k-connectedness
+        (te : H-level-truncations-exist)
+       where
+
+ open H-level-truncations-exist te
+
+ _is_connected : 𝓤 ̇ → ℕ → 𝓤 ̇
+ X is k connected = is-contr (∥ X ∥[ k ])
+
+ map_is_connected : {X : 𝓤 ̇} {Y : 𝓥 ̇} → (f : X → Y) → ℕ → 𝓤 ⊔ 𝓥 ̇
+ map f is k connected = (y : codomain f) → (fiber f y) is k connected
+
+\end{code}
+
+We now characterize 1-connected types as inhabited types and 1-connected maps
+as surjections.
+
+\begin{code}
+
+ open propositional-truncations-exist pt
+
+ one-connected-is-inhabited : {X : 𝓤 ̇}
```

Comment:
Clearer: `inhabited-if-one-connected`. Please follow this convention everywhere.